cesql/spec.md

```diff
@@ -197,8 +207,13 @@ Unless otherwise specified, every attribute and extension MUST be represented by
 Through implicit type casting, the user can convert the addressed value instances to _Integer_ and
```

Comment:
"... MUST be represented by the _String_ type as its initial type"...  I'm curious what this really means and why it's here. Does the user who is creating the expression need to do something with this info? Or is this for an implementation/SDK ? And if so, what do they do with it? If someone defines an extension called myint (an int), why would it matter if it starts out (internally) as an int or a string if it will be type cast as needed?

The spec says: The primary output of a CESQL expression evaluation is always a boolean, an integer or a string.

Does this mean that an expression of just `myint` MUST result in a string? And if so, it's kind of interesting that the spec doesn't define an int() function to allow people to covert it back to an int if that's what they want to result to be.

Reply:
I'm still not following. If I define an extension as an int, why must it be converted to a string ?
Especially if the cesql expression if `myint = 5` ?

Reply:
Let me ask this... what breaks if you just remove these sentences and this notion of "initial type"? Things will get converted automatically when needed. I'm just not seeing the point of (or the meaning of) the phrase "initial type".

Thinking about this more... is this exposing some implementation detail because the SDKs convert all attributes to strings initially? If so, then it really shouldn't be in the spec since the spec should focus on the cesql language syntax and resulting semantics, not how the code goes from "syntax" to "result".

Reply:
How about "If the value of the attribute or extension is not one of the primitive CESQL types, it MUST be represented by the _String_ type"